app/models/spree/user_decorator.rb

```diff
@@ -1,20 +1,22 @@
 module Spree::UserDecorator
 
   def self.prepended(base)
-    base.after_create :create_mollie_customer
+    base.after_commit :ensure_mollie_customer, on: %i[create update]
   end
 
 
-  def create_mollie_customer
+  def ensure_mollie_customer
+    return if mollie_customer_id.present?
```

Comment:
It appears that the mollie_customer_id can be undefined. Either by the migration not being executed or because of some other reason.

Reply:
So it would be best to accept that it could be missing?

Reply:
Or we could move it down to below the active?-check